### PR TITLE
feat: EIP initial implementation

### PR DESCRIPTION
**Pull request by ktechmidas**
## Issue being fixed or feature implemented
Change public IP's to elastic IP addresses for HPMNs. This ensures HPMNs will keep their IP address through rebooting/changing things/etc

## What was done?
New blocks for EIPs for both ARM/AMD, turn public_ip_allocation off for those nodes and add an EIP instead.

## How Has This Been Tested?
devnet-eip

## Breaking Changes
~None, AFAIK~
Requires Terraform version 1.4.4 due to bug in previous versions.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: terraform/aws/main.tf [hunk 1]
```diff
@@ -240,22 +240,22 @@ resource "random_shuffle" "dns_ips" {
   input        = concat(
     aws_instance.masternode_amd.*.public_ip,
     aws_instance.masternode_arm.*.public_ip,
-    aws_instance.hp_masternode_amd.*.public_ip,
-    aws_instance.hp_masternode_arm.*.public_ip
+      aws_eip.hpmn_arm_eip.*.public_ip,
+      aws_eip.hpmn_amd_eip.*.public_ip
```
Comment (strophy) [inline terraform/aws/main.tf:244]: It's definitely necessary to reference the EIP itself here right? It's not possible to reference the instance instead, and let the graph figure out that the association is necessary before it gets to this step? After all the EIP *is* the `public_ip` associated with the instance.

Comment (ktechmidas) [reply terraform/aws/main.tf:244]: Definitely necessary unfortunately, the public_ip within the instance itself is the one AWS gives it before the EIP, not the EIP.

File: terraform/aws/instances.tf [hunk 2]
```diff
@@ -317,6 +336,29 @@ resource "aws_instance" "hp_masternode_arm" {
...
+}
+
+
+output "hp_masternode_arm_eip" {
+  value = aws_eip.hpmn_arm_eip.*.public_ip
+}
+
+output "hp_masternode_amd_eip" {
+  value = aws_eip.hpmn_amd_eip.*.public_ip
+}
```
[the remark this answers is not in the document]
Comment (ktechmidas) [reply terraform/aws/instances.tf]: Was for debugging, removed now.

**Review by strophy — APPROVED** [review]
Works for me :rocket:

**Pull request merged**